backend/src/Entity/UserChecklist.php

```diff
@@ -20,30 +20,99 @@ class UserChecklist
     private $id;
 
     /**
-     * @ORM\Column(type="boolean")
+     * @ORM\OneToMany(targetEntity=User::class, mappedBy="userChecklist")
      */
-    private $isComplete;
+    private $user;
 
+    /**
+     * @ORM\OneToMany(targetEntity=ChecklistItem::class, mappedBy="userChecklist")
+     */
+    private $checklistItem;
+
+    /**
+     * @ORM\Column(type="boolean")
+     */
+    private $isChecked;
 
     public function __construct()
     {
-        $this->user_id = new ArrayCollection();
-        $this->checklist_id = new ArrayCollection();
+        $this->user = new ArrayCollection();
+        $this->checklistItem = new ArrayCollection();
     }
 
     public function getId(): ?int
     {
         return $this->id;
     }
 
-    public function getIsComplete(): ?bool
+    /**
+     * @return Collection|User[]
+     */
+    public function getUser(): Collection
+    {
+        return $this->user;
+    }
+
+    public function addUser(User $user): self
+    {
+        if (!$this->user->contains($user)) {
+            $this->user[] = $user;
+            $user->setUserChecklist($this);
+        }
+
+        return $this;
+    }
+
+    public function removeUser(User $user): self
+    {
+        if ($this->user->removeElement($user)) {
+            // set the owning side to null (unless already changed)
+            if ($user->getUserChecklist() === $this) {
+                $user->setUserChecklist(null);
+            }
+        }
```

Comment:
You can combine both `if` statements

```suggestion
        if ($this->user->removeElement($user) && $user->getUserChecklist() === $this) {
            // set the owning side to null (unless already changed)
            $user->setUserChecklist(null);
        }
```

or

```suggestion
        if ($this->user->removeElement($user) AND $user->getUserChecklist() === $this) {
            // set the owning side to null (unless already changed)
            $user->setUserChecklist(null);
        }
```